### ISSUE TITLE
`build` should be able to create a Windows image

### ISSUE THREAD

**Issue by natalieparellano** [buildpacks/pack#470]
### Description
`pack build` currently only creates Linux images. 

### Proposed solution
`pack` (compiled for Linux and Mac) and `pack.exe` (compiled for Windows) should be able to create Windows images.

### Additional context
* This work should follow / is blocked by https://github.com/buildpacks/pack/issues/469
* Leaving buildpackages out of scope for now due to their complexity
* This issue may / is intended to drive out further issues in the [lifecycle](https://github.com/buildpacks/lifecycle) for Windows compatibility
* When `pack` is compiled for Linux or Mac, it is assumed that the docker daemon being targeted is running on Windows
* When the default stack is Linux, should `--run-image` be required for Windows builds?

### Possible acceptance criteria
* Given a builder that is a Windows image
* `pack build windows-app-image --builder windows-builder --buildpack minimal-windows-buildpack --path windows-app` should succeed with and without `--publish`
* `pack.exe build windows-app-image --builder windows-builder --buildpack minimal-windows-buildpack --path windows-app` should succeed with and without `--publish`
* `docker run windows-app-image` should succeed and the app should start correctly
